### ISSUE TITLE
[Question] how to use expect.any in playwright

### ISSUE THREAD

**Issue by FrankWang117** [microsoft/playwright#12814]
I want to :
```typescript
expect(id).toEqual(expect.any(String))
```

**Comment by aslushnikov** [comment]
You can do 

```js
expect(typeof id).toBe('string')
```

**Comment by pavelfeldman** [comment]
This looks like a regression from https://github.com/microsoft/playwright/pull/12403, does it?

**Comment by FrankWang117** [comment]
@aslushnikov maybe.
but if I want to compare object:
```typescript
expect(request).toEqual({
    foo: 1,
    id: expect.any(String)
})
```
how to do?
Now I use:
```typescript
 expect(request).toMatchObject({
    foo: 1
 });
expect(typeof request.id).toBe('string');
```

**Comment by pavelfeldman** [comment]
```
import { test, expect } from '@playwright/test';

test('foo', async () => {
  expect('id').toEqual(expect.any(String));
});
```

works for me, it is just that the types aren't right, we'll fix it right away.

**Comment by FrankWang117** [comment]
@pavelfeldman ok,thanks a lot.

**Comment by pavelfeldman** [comment]
Fixed, awaiting the 1.20.1 push.
